recipes/repository.rb

```diff
@@ -48,15 +46,26 @@
   end
 when 'redhat', 'centos'
   # CentOS 6 and 7 have Gluster in the Storage SIG instead of a gluster hosted repo
-  repo_url = if node['platform_version'].to_i > 5
-               "http://mirror.centos.org/centos/$releasever/storage/$basearch/gluster-#{node['gluster']['version']}/"
-             else
-               "http://download.gluster.org/pub/gluster/glusterfs/#{node['gluster']['version']}/LATEST/EPEL.repo/epel-$releasever/$basearch/"
-             end
+  if node['platform_version'].to_i > 5
+    if Chef::VersionConstraint.new('>= 3.9').include?(node['gluster']['version'])
+      subdomain = 'buildlogs'
+      gpg_url = nil
+    else
+      subdomain = 'mirror'
+      gpg_url = 'https://raw.githubusercontent.com/CentOS-Storage-SIG/centos-release-storage-common/master/RPM-GPG-KEY-CentOS-SIG-Storage'
```

Comment:
Not sure that i am a fan of this being the source of the key.
The key is provided by the centos-release-storage-common package in the centos extras repo